rest/index.py

```diff
@@ -468,6 +481,433 @@ def createIncident():
+                "error": "user.identityConfidence not one of {}".format(
+                    " or ".join(valid_confidences)
+                )
+            }
+        )
+        return response
+
+    details = {
+        "triage": {"user": req["user"], "response": req["response"]},
+        "identityConfidence": req["identityConfidence"],
+    }
+
+    modified_count = 0
+
+    modified_count += alerts.update_one(
+        {"esmetadata.id": req["alert"]}, {"$set": {"status": req["status"]}}
+    ).modified_count
+
+    modified_count += alerts.update_one(
+        {"esmetadata.id": req["alert"]}, {"$set": {"details": details}}
+    ).modified_count
+
+    if modified_count < 2:
+        response.status = StatusCode.BAD_REQUEST
+        response.body = json.dumps({"error": "Alert not found"})
+        return response
+
+    response.status = StatusCode.OK
+    response.body = json.dumps({"error": None})
+
+    return response
+
+
+@get("/alerttriagechain")
+@get("/alerttriagechain/")
+def retrieve_duplicate_chain():
+    """Search for a `Duplicate Chain` storing information about duplicate
+    alerts triggered by the same user's activity.  These chains track such
+    duplicate alerts so that the triage bot does not have to message a user
+    on Slack for each such alert within some period of time.
+
+    Requests are expected to include the following query parameters.
+
+        * `"alert": str` is the "label" for the alert, signifying which of the
+        supported alerts the user in question triggered.
+        * `"user": str` is the email address of the user contacted.
+
+    This function writes back a response containing the following JSON.
+
+    ```
+    {
+        "error": Optional[str],
+        "identifiers": List[str],
+        "created": str,
+        "modified": str
+    }
+    ```
+
+    Here,
+        * `"error"` will contain a string message if any error occurs performing
+        a lookup.  If such an error occurs, `"identifiers"` will be an empty
+        list.
+        * `"identifiers"` is a list of IDs of alerts stored under the chain.
+        * `"created"` is the date & time at which the chain was created.
+        * `"modified"` is the date & time at which the chain was last modified.
+
+    Both the `"created"` and `"modified"` fields represent UTC timestamps and
+    are formatted like `YYYY/mm/dd HH:MM:SS`.
+    """
+
+    response.content_type = "application/json"
+
+    initConfig()
+
+    mongo = MongoClient(options.mongohost, options.mongoport)
+    dupchains = mongo.meteor[DUP_CHAIN_DB]
+
+    def _error(msg):
+        return json.dumps({
+            "error": msg,
+            "identifiers": [],
+            "created": toUTC(datetime.utcnow()).isoformat(),
+            "modified": toUTC(datetime.utcnow()).isoformat(),
+        })
+
+    query = {"alert": request.query.alert, "user": request.query.user}
+
+    if query["alert"] is None or query["user"] is None:
+        response.status = StatusCode.BAD_REQUEST
```

Comment:
I think what you want to be testing for here [is an empty string, not `None`](https://bottlepy.org/docs/dev/api.html#bottle.FormsDict) because `FormsDict` returns empty strings on missing values.